packages/react-native-tracer-provider/android/src/main/java/io/embrace/tracerprovider/TracerProviderModule.kt

```diff
@@ -0,0 +1,288 @@
+    private val SPAN_STATUS_MESSAGE_KEY = "message";
+
+    private fun stringListFromReadableArray(array: ReadableArray) :List<String> {
+        val list: MutableList<String> = mutableListOf();
+
+        for(i in 0..<array.size()) {
+            list.add(array.getString(i))
+        }
+        return list
+    }
+
+    private fun doubleListFromReadableArray(array: ReadableArray) :List<Double> {
+        val list: MutableList<Double> = mutableListOf();
+
+        for(i in 0..<array.size()) {
+            list.add(array.getDouble(i))
+        }
+        return list
+    }
+
+    private fun booleanListFromReadableArray(array: ReadableArray) :List<Boolean> {
+        val list: MutableList<Boolean> = mutableListOf();
+
+        for(i in 0..<array.size()) {
+            list.add(array.getBoolean(i))
+        }
+        return list
+    }
+
+    private fun attributesFromReadableMap(map: ReadableMap) : Attributes {
+        val it = map.keySetIterator()
+        val builder = Attributes.builder()
+
+        while(it.hasNextKey()) {
+            val key = it.nextKey()
+            val type = map.getType(key);
+
+            when (type) {
+                ReadableType.Boolean -> builder.put(key, map.getBoolean(key))
+                ReadableType.Number -> builder.put(key, map.getDouble(key))
+                ReadableType.String -> builder.put(key, map.getString(key) ?: "")
+                ReadableType.Array -> {
+                    val array = map.getArray(key)
+                    val arrayType = array?.getType(0);
+                    when (arrayType) {
+                        ReadableType.Boolean -> builder.put(AttributeKey.booleanArrayKey(key), booleanListFromReadableArray(array))
+                        ReadableType.Number -> builder.put(AttributeKey.doubleArrayKey(key), doubleListFromReadableArray(array))
+                        ReadableType.String -> builder.put(AttributeKey.stringArrayKey(key), stringListFromReadableArray(array))
+                        else -> continue
+                    }
+                }
+                else -> continue
+            }
+        }
+
+
+        return builder.build();
+    }
+
+    private fun spanContextFromReadableMap(map: ReadableMap) : SpanContext {
+        val traceID = map.getString(SPAN_CONTEXT_TRACE_ID_KEY) ?: ""
+        val spanID = map.getString(SPAN_CONTEXT_SPAN_ID_KEY) ?: ""
+
+        val spanContext = SpanContext.create(
+            traceID,
+            spanID,
+            TraceFlags.getDefault(),
+            TraceState.getDefault())
+
+        return spanContext
+    }
+
+    private fun spanContextToWritableMap(spanContext: SpanContext?) : WritableMap{
+        val map = this.writableMapBuilder.build()
+
+        map.putString(SPAN_CONTEXT_TRACE_ID_KEY, spanContext?.traceId ?: "")
+        map.putString(SPAN_CONTEXT_SPAN_ID_KEY, spanContext?.spanId ?: "")
+
+        return map
+    }
+
+    init {
+        // TODO replace with Embrace OTEL provider
```

Comment:
Is this comment up to date?

Reply:
yep that's part of why this is still in draft this is still using just the OTEL SDK for prototyping purposes it will need to be updated to use the actual Embrace one